### DISCUSSION TITLE
Revoke E-Mail notifications prints the variable name instead of their content

### DISCUSSION THREAD

**Discussion by bugz8unny69** [Keyfactor/ejbca-ce#283] [Q&A]
Hello, community,

This could be me miss-understanding the documentation [here](https://doc.primekey.com/ejbca/ejbca-operations/ejbca-ca-concept-guide/end-entities-overview/end-entity-profiles-overview/e-mail-notifications). But for whatever reason unbeknownst to me, revoked E-Mails do not print the variable contents. 

Recipient E-Mail Example:
```
2023-05-13 07:02:04+00:00

Hi testuser,

You are being notified to inform you that your login certificate to xxxxx has been revoked. 

Serial: ${revokedCertificate.CERTSERIAL}
Reason: ${revokedCertificate.REVOCATIONREASON}

If you believe this has been done in error, please contact us at xxxxx.

Regards,
SuperAdmin
xxxxx
```

Workflow:
1. Create an End Entity Profile
2. Enable E-Mail Notification for STATUSREVOKED
3. Create an E-Mail template with any of the $[revokedCertificate.**} variables.

Expected:

Prints the variable contents

Problem:

Prints variable names instead.

Regards,
Horace

**Comment by primetomas** [comment]
I made a quick test and it works for me. It only works when revoking a certificate though. 
I tested now with the same variables both when adding the end entity, generating the certificate, and revoking the certificate (revoking the cert, not revoking the end entity).

When issuing the cert I get:
The serial number of the revoked certificate: ${revokedCertificate.CERTSERIAL} 
The date the revoked certificate will expire: ${revokedCertificate.EXPIREDATE}
The certificate subject DN: ${revokedCertificate.CERTSUBJECTDN}
The certificate issuer DN: ${revokedCertificate.CERTISSUERDN}
The new status of the certificate (Active/Revoked): ${revokedCertificate.REVOCATIONSTATUS}
The new revocation reason of the certificate (unspecified, keyCompromise, ...): ${revokedCertificate.REVOCATIONREASON}

When revoking the cert I get:
The serial number of the revoked certificate: 73617A510E6D2EE936BEBAF155D7F079 
The date the revoked certificate will expire: 2025-05-22 08:30:57+02:00
The certificate subject DN: CN=email1
The certificate issuer DN: CN=Management CA,O=PK-DM,C=SE
The new status of the certificate (Active/Revoked): Revoked
The new revocation reason of the certificate (unspecified, keyCompromise, ...): certificateHold

**Comment by bugz8unny69** [comment]
`not revoking the end entity`

This is the key here. Why is there a distinction? Why revoking the certificate itself, different from revoking an end entity? Aren't they the same? What would be the proper procedure for revoking a client certificate? Revoke the certificate and leave the end entity active?

Regards.

**Comment by primetomas** [comment]
An end entity can have multiple different certificates. Also revoking an end entity will prevent it from renewing certificates (if enables in certain protocols). The end entity represents a user/device/server you can say. 
Depending on your use case, but I would say yes, the most common revocation scenario is that you revoke the certificate but leave the end entity as is (typically in a "generated" state). But it depends on use case, protocols used etc. For example you create an end entity for 3GPP eNodeBs so they can automatically enroll for an operator certificate when connected to the network. If the eNodeB is reported stolen you can revoke the end entity in order to prevent this eNodeB to get any certificates (the initial one or automatically renewing if it did get the first one).
